### PR TITLE
[Optimization](Variant) Using correct column name to go fast execute …

### PR DESCRIPTION
**Pull request by eldenmoon**
…path  in executing match expressions

The correct column name should be like `CAST value.name(Nullable(Variant)) TO Nullable(String)` since the predicate is `cast(value['name'] as string) match 'xxx'`

Query : SELECT value FROM test.records WHERE entity_id = "xxxx"  value['text3']  MATCH_ALL ' 就是 射手 男 的 克'  OR ( value['text83']  MATCH_ALL ' 考试 的 孩子 们 有 福利 ' ) OR (  value['text15']  MATCH_ALL '得 台湾 女星 许' )  LIMIT 0, 100

from 3s ~ 0.3s

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by zzzxl1993 — APPROVED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -2692,13 +2715,16 @@ void SegmentIterator::_calculate_pred_in_remaining_conjunct_root(
 
     auto node_type = expr->node_type();
     if (node_type == TExprNodeType::SLOT_REF) {
+        auto slot_expr = std::dynamic_pointer_cast<doris::vectorized::VSlotRef>(expr);
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2718]: need if (slot_expr != nullptr) ?

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:2718]: if (node_type == TExprNodeType::SLOT_REF)  ， it must be doris::vectorized::VSlotRef

**Pull request merged**